File: test_overpass.py
import requests
import json

def test_overpass_api():
    print("OpenStreetMap Overpass API Testi Başlıyor...")
    
    # Overpass API endpoint
    url = "http://overpass-api.de/api/interpreter"
    
    # İstanbul için örnek bir sorgu oluşturuyoruz.
    # Tüm Türkiye'yi çekmek uzun sürebilir, test için İstanbul'u seçtik.
    # office=accountant veya office=tax_advisor olan yerleri arıyoruz.
    query = """
    [out:json][timeout:50];
    area["name"="İstanbul"]->.searchArea;
    (
      node["office"="accountant"](area.searchArea);
      way["office"="accountant"](area.searchArea);
      relation["office"="accountant"](area.searchArea);
      
      node["office"="tax_advisor"](area.searchArea);
      way["office"="tax_advisor"](area.searchArea);
      relation["office"="tax_advisor"](area.searchArea);
    );
    out tags;
    """
    
    print("Sorgu API'ye gönderiliyor (Bu işlem birkaç saniye sürebilir)...")
    response = requests.post(url, data={'data': query})
    
    if response.status_code == 200:
        data = response.json()
        elements = data.get('elements', [])
        
        print(f"\nToplam {len(elements)} adet mali müşavir / muhasebeci kaydı bulundu (İstanbul).")
        
        # Web sitesi olan kayıtları filtreleyelim
        with_website = [e for e in elements if 'tags' in e and 'website' in e['tags']]
        
        print(f"Bunlardan {len(with_website)} tanesinin web sitesi sisteme kayıtlı.\n")
        print("Web sitesi olan ilk 5 kaydın detayları:")
        print("-" * 50)
        
        for i, el in enumerate(with_website[:5]):
            tags = el.get('tags', {})
            name = tags.get('name', 'İsimsiz İşletme')
            website = tags.get('website', 'Yok')
            phone = tags.get('phone', tags.get('contact:phone', 'Yok'))
            
            print(f"{i+1}. İşletme Adı: {name}")
            print(f"   Web Sitesi: {website}")
            print(f"   Telefon: {phone}")
            print("-" * 50)
            
    else:
        print(f"Hata oluştu! Durum Kodu: {response.status_code}")
        print(response.text)

if __name__ == "__main__":
    test_overpass_api()

File: test_overpass.js
const query = `
[out:json][timeout:25];
(
  node["office"="accountant"](40.9, 28.9, 41.1, 29.1);
  way["office"="accountant"](40.9, 28.9, 41.1, 29.1);
  relation["office"="accountant"](40.9, 28.9, 41.1, 29.1);
  
  node["office"="tax_advisor"](40.9, 28.9, 41.1, 29.1);
  way["office"="tax_advisor"](40.9, 28.9, 41.1, 29.1);
  relation["office"="tax_advisor"](40.9, 28.9, 41.1, 29.1);
);
out body;
>;
out skel qt;
`;

async function testOverpassApi() {
    console.log("OpenStreetMap Overpass API Testi Başlıyor...");
    console.log("Alternatif API sunucusuna bağlanılıyor (Kumi Systems)...");

    try {
        const response = await fetch("https://overpass.kumi.systems/api/interpreter", {
            method: "POST",
            headers: {
                "Content-Type": "application/x-www-form-urlencoded",
                "User-Agent": "MaliGorus-Test/1.0",
                "Accept": "application/json"
            },
            body: `data=${encodeURIComponent(query)}`
        });

        if (!response.ok) {
            const errText = await response.text();
            throw new Error(`HTTP error! status: ${response.status}, message: ${errText}`);
        }

        const data = await response.json();
        const elements = data.elements || [];

        console.log(`\nToplam ${elements.length} adet harita objesi bulundu.`);

        // Sadece web sitesi olan muhasebeci/mali müşavirleri filtrele
        const withWebsite = elements.filter(el => 
            el.tags && 
            (el.tags.office === "accountant" || el.tags.office === "tax_advisor") && 
            el.tags.website
        );
        
        console.log(`Bunlardan ${withWebsite.length} tanesinin web sitesi sisteme kayıtlı.\n`);
        
        if(withWebsite.length > 0) {
            console.log("Web sitesi olan ilk 5 kaydın detayları:");
            console.log("-".repeat(50));

            withWebsite.slice(0, 5).forEach((el, index) => {
                const tags = el.tags || {};
                const name = tags.name || "İsimsiz İşletme";
                const website = tags.website || "Yok";
                const phone = tags.phone || tags["contact:phone"] || "Yok";

                console.log(`${index + 1}. İşletme Adı: ${name}`);
                console.log(`   Web Sitesi: ${website}`);
                console.log(`   Telefon: ${phone}`);
                console.log("-".repeat(50));
            });
        } else {
            console.log("Bu küçük alanda web sitesi eklenmiş bir ofis bulunamadı. Aramayı tüm Türkiye'ye genişletebiliriz.");
        }

    } catch (error) {
        console.error("Hata oluştu:", error);
    }
}

testOverpassApi();
